src/analysis/retail/mage/frost/talents/Flurry.tsx

```diff
@@ -2,30 +2,129 @@ import SPELLS from 'common/SPELLS';
   static dependencies = {
+    enemies: Enemies,
     spellUsable: SpellUsable,
   };
+
+  protected enemies!: Enemies;
   protected spellUsable!: SpellUsable;
+  flurryEvents: FlurryEvent[] = [];
 
   constructor(props: Options) {
     super(props);
     this.addEventListener(
       Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.BRAIN_FREEZE_BUFF),
       this._gainCharge,
     );
+    this.addEventListener(
+      Events.cast.by(SELECTED_PLAYER).spell(TALENTS.FLURRY_TALENT),
+      this.onFlurryCast,
+    );
+    this.addEventListener(Events.fightend, this.analyzeFlurry);
+  }
+
+  onFlurryCast(event: CastEvent) {
+    const damage: DamageEvent | undefined = GetRelatedEvent(event, 'SpellDamage');
+    const enemy = damage && this.enemies.getEntity(damage);
+    const icicles =
+      this.selectedCombatant.getBuff(SPELLS.ICICLES_BUFF.id, event.timestamp)?.stacks || 0;
+    const buffRemove = GetRelatedEvent(event, 'BuffRemove');
+    const flurryEvent = new FlurryEvent(event, damage, enemy, icicles, buffRemove !== undefined);
+    this.flurryEvents.push(flurryEvent);
+  }
+
+  analyzeFlurry(): { label: string; color: string; value: number }[] {
+    const withBrainFreeze = this.flurryEvents.filter((flurry) => flurry.brainFreeze).length;
+    const withoutBrainFreezeGood = this.flurryEvents.filter((flurry) =>
+      flurry.noBfGoodCast(),
+    ).length;
+    const withoutBrainFreezeBad = this.flurryEvents.filter((flurry) => flurry.noBfBadCast()).length;
+
+    const flurryCasts: { label: string; color: string; value: number }[] = [];
+
+    flurryCasts.push({
+      label: 'with Brain Freeze (BF)',
+      color: colors[0],
+      value: withBrainFreeze,
+    });
+    flurryCasts.push({
+      label: 'without BF, 2 or more icicles',
+      color: colors[1],
+      value: withoutBrainFreezeGood,
+    });
+    flurryCasts.push({
+      label: 'without BF, 0/1 icicles',
+      color: colors[2],
+      value: withoutBrainFreezeBad,
+    });
+
+    return flurryCasts;
   }
 
   _gainCharge() {
     if (this.spellUsable.isOnCooldown(TALENTS.FLURRY_TALENT.id)) {
       this.spellUsable.reduceCooldown(TALENTS.FLURRY_TALENT.id, REDUCTION_MS);
     }
   }
+
+  get guideSubsection(): JSX.Element {
+    const flurry = <SpellLink spell={TALENTS.FLURRY_TALENT} />;
+    const glacialSpike = <SpellLink spell={TALENTS.GLACIAL_SPIKE_TALENT} />;
+    const frostbolt = <SpellLink spell={SPELLS.FROSTBOLT} />;
+    const brainFreeze = <SpellLink spell={TALENTS.BRAIN_FREEZE_TALENT} />;
+    const icicles = <SpellLink spell={SPELLS.MASTERY_ICICLES} />;
+
+    const explanation = (
+      <>
+        {flurry} usage is important to make sure you can shatter as much {glacialSpike} as you can.
+        You should only hold it after a {frostbolt} hardcast if you don't have {brainFreeze} and you
+        have 0 or 1 {icicles}. <br />
+        <small>
+          At 2 or more {icicles} you are allways able to shatter {glacialSpike}: <br />
+        </small>
```

Comment:
would recommend trying `<p>` for these instead of `<br />`. the default spacing may not be what you want, but generally should prefer block/containing elements instead of `br`

```suggestion
        <p>{flurry} usage is important to make sure you can shatter as much {glacialSpike} as you can.
        You should only hold it after a {frostbolt} hardcast if you don't have {brainFreeze} and you
        have 0 or 1 {icicles}. </p>
        <p><small>
          At 2 or more {icicles} you are always able to shatter {glacialSpike}:
        </small></p>
```